### PR TITLE
Fix field promulgated for audit log.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1062/
Test PASSed.

**Comment by fabricematrat** [comment]
QA:
Choose a charm in your charmstore and then promulgate and un promulgate.
to promulgate use for example
http put admin:example-passwd@localhost:8080/v4/~cf-charmers/trusty/cf-mysql-broker-23/promulgate <promulgate.json Content-Type:application/json
where promulgate.json is {"Promulgated":true}

you should get entries in the audit.lo like the following:
 {"time":"2015-06-19T06:09:46.540496636+02:00","user":"admin","op":"set-promulgated","entity":"cs:~cf-charmers/trusty/cf-mysql-broker-23","promulgated":true}

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1063/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1064/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1065/
Test PASSed.

**Comment by mhilton** [comment]
:+1: LGTM QA OK

File: audit/audit.go [hunk 1]
```diff
@@ -16,9 +16,10 @@ const (
 	// Required fields: Entity, ACL
 	OpSetPerm Operation = "set-perm"
 
-	// OpSetPromulgated represents the promulgation on an entity.
-	// Required fields: Entity, Promulgated
-	OpSetPromulgated Operation = "set-promulgated"
+	// OpSetPromulgate, OpSetUnPromulgate represent the promulgation on an entity.
```
Comment (rogpeppe) [inline audit/audit.go]: I'd suggest OpPromulgate and OpUnpromulgate (the "Set" is redundant).

```
// OpPromulgate represents the promulgation of an entity.
OpPromulgate = "promulgate"

// OpUnpromulgate represents the unpromulgation of an entity.
OpUnpromulgate = "unpromulgate"
```

File: internal/v4/api_test.go [hunk 2]
```diff
@@ -2557,12 +2557,21 @@ func (s *APISuite) TestPromulgate(c *gc.C) {
 			ref.Series = "trusty"
 			ref.Revision = 0
 
-			c.Assert(calledEntities, jc.DeepEquals, []audit.Entry{{
-				User:        test.expectUser,
-				Op:          audit.OpSetPromulgated,
-				Entity:      ref,
-				Promulgated: test.expectPromulgate,
-			}})
+			var e []audit.Entry
```
Comment (rogpeppe) [inline internal/v4/api_test.go]: perhaps:

```
 e := audit.Entry{
      User: test.expectUser,
      Op: audit.OpUnpromulgate,
      Entity: ref,
 }
 if test.expectPromulgate {
       e.Op = audit.OpPromulgate
 }
 c.Assert(calledEntities, gc.DeepEquals, []audit.Entry{e})
```

?

**Comment by rogpeppe** [comment]
LGTM modulo the above suggestions, thanks.

File: audit/audit.go (hunk 1 above)
Comment (fabricematrat) [inline audit/audit.go]: Done.

File: internal/v4/api_test.go (hunk 2 above)
Comment (fabricematrat) [inline internal/v4/api_test.go]: Done.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1068/
Test PASSed.

**Comment by fabricematrat** [comment]
:shipit: thanks @mhilton && @rogpeppe

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**